### PR TITLE
fuge pull repo functionality

### PR DESCRIPTION
**Pull request by mihaidma**

### CODE REVIEW

File: lib/gitPuller.js [hunk 1]
```diff
@@ -45,29 +41,46 @@ module.exports = function (config) {
...
+          exitCb(child.pid, toExec, code)
+        }
       }
     })
     cb(null, child)
   }
 
 
 
+  var pull = function (name, cb) {
```
Comment (mcdonnelldean) [inline lib/gitPuller.js]: Since we have named this stuff generate now, does this need to change?

File: test/pullerTest.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+'use strict'
```
Comment (mcdonnelldean) [inline test/pullerTest.js]: Tests!!! Awesome :D

**Comment by mihaidma** [comment]
@mcdonnelldean changes done, added also full url support

**Comment by mcdonnelldean** [comment]
LGTM but let me check tomorrow too because we are getting a failure in AppVeyor.

**Pull request merged**